src/packages/auth/src/authentication/apollo/plugin.ts

```diff
@@ -70,9 +76,35 @@ export const authApolloPlugin = (
 
 			// If verification fails then set this flag
 			let tokenVerificationFailed = false;
-
-			// Case 1. No auth header or it has expired.
-			if (!authHeader || isExpired(authHeader)) {
+			let apiKeyVerificationFailedMessage: string | undefined = undefined;
+
+			if (apiKeyHeader && apiKeyDataProvider) {
+				// Case 1. API Key auth header found.
+				logger.trace('X-API-Key header found checking validity.');
+
+				const [key, secret] = Buffer.from(apiKeyHeader, 'base64').toString('utf-8').split(':');
+
+				const apiKey = await apiKeyDataProvider?.findOne({
+					key,
+				} as D);
+
+				if (!apiKey || !apiKey.secret) {
+					apiKeyVerificationFailedMessage = 'Bad Request: API Key Authentication Failed. (E0001)';
+				} else if (apiKey.revoked) {
+					apiKeyVerificationFailedMessage = 'Bad Request: API Key Authentication Failed. (E0002)';
+				} else if (await verifyPassword(secret, apiKey.secret)) {
+					contextValue.user = new UserProfile({
```

Comment:
Why do we new up a UserProfile here? Not sure of another way to do this but every request will create a new profile in memory?

Reply:
yes, and that is ok as it is used by the permission system.